src/attribute_.h

```diff
@@ -17,6 +17,70 @@ class Attribute_ : public Attribute {
     // escalated to the user.
     using Validator = std::function<std::string(T)>;
 
+    /** The constructors for Attribute_ and also for DynAttribute_ follow
+     * the a common pattern:
+     *
+     *   - The first argument is the owner object
+     *
+     *   - The second argument is the name displayed to the user
+     *
+     *   - The third argument the reading behaviour of the attribute:
+     *     * for a plain Attribute_ this is simply the initial value
+     *     * for a DynAttribute_ this is a member of the owner (or a lambda)
+     *       that returns the value of the attribute.
+     *
+     *   - The fourth argument describes the writing behaviour of the
+     *     attribute: If the fourth argument is absent, the attribute is
+     *     read-only.
+     *     * The fourth argument for an Attribute_ is a member of owner (or
+     *       lambda) that validates a new value of the attribute and returns an
+     *       error message if the new value is not acceptable for this
+     *       attribute (E.g. because a name is already taken or does not
+     *       resepct a certain format).
+     *     * The fourth argument of a DynAttribute_ is a member of owner (or
+     *       lambda) that internally processes the new value (e.g. parsing) and
+     *       returns an error message if the new value is not acceptable.
+     */
+
+    //! A read-only attribute of owner of type T
+    Attribute_(Object* owner, const std::string &name, const T &payload)
+        : Attribute(name, false)
+        , payload_ (payload)
+    {
+        // the following will call Attribute::setOwner()
+        // maybe this should be changed at some point,
+        // e.g. when we got rid of Object::wireAttributes()
+        owner->addAttribute(this);
+    }
+
+    //! A writable attribute of owner of type T
+    template <typename Owner>
+    Attribute_(Owner* owner, const std::string &name, const T &payload,
+              std::string(Owner::*validator)(T))
+        : Attribute(name, true)
+        , validator_(std::bind(validator, owner, std::placeholders::_1))
+        , payload_ (payload)
+    {
+        // the following will call Attribute::setOwner()
+        // maybe this should be changed at some point,
+        // e.g. when we got rid of Object::wireAttributes()
+        owner->addAttribute(this);
+    }
+    //! A writable attribute of owner of type T
+    Attribute_(Object* owner, const std::string &name, const T &payload,
+              Validator validator)
+        : Attribute(name, true)
+        , validator_(validator)
+        , payload_ (payload)
+    {
+        // the following will call Attribute::setOwner()
+        // maybe this should be changed at some point,
+        // e.g. when we got rid of Object::wireAttributes()
+        owner->addAttribute(this);
```

Comment:
Just a thought: Maybe `registerAttribute` would be an (even) more descriptive name for that method?

Apart from that, I agree with the comment above. I think/hope `setOwner` as a public method can be fully replaced with initializers in the attribute ctors.

Reply:
Yes, and then it's clear that `registerAttribute` does not call `Attribute::setOwner`.